### PR TITLE
New Code Example: Training T5 on abstractive summarization using Hugging Face Transformers

### PR DESCRIPTION
**Pull request by Sreyan88**
This code example shows how to fine-tune T5 on the abstraction summarization task using Hugging Face Transformers. Additionally, it also shows best practices for Hugging Face Datasets.

### CODE REVIEW

**Comment by Sreyan88** [comment]
Hi @fchollet,

This is part of my GSoC 2022 contribution. Hope you can review it and comment!

Thank You so much!

File: examples/nlp/summarization.py [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+"""
+
+"""
+Now we will define our optimizer and compile the model. The loss calculation is handled
+internally and so we need not worry about that!
+"""
+
+from transformers import AdamWeightDecay
+
+optimizer = AdamWeightDecay(learning_rate=LEARNING_RATE, weight_decay_rate=WEIGHT_DECAY)
```
Comment (fchollet) [inline examples/nlp/summarization.py:281]: Q: does it work for you to use `keras.optimizers.experimental.AdamW`?

File: examples/nlp/summarization.py [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+    return result
+
+
+"""
+Now we can finally start training our model!
+"""
+
+from transformers.keras_callbacks import KerasMetricCallback
+
+metric_callback = KerasMetricCallback(
```
Comment (fchollet) [inline examples/nlp/summarization.py:327]: Q: instead of using a callback for Rouge, can you use the KerasNLP metric? You'd just pass it to `compile()`. https://keras.io/api/keras_nlp/metrics/

**Review by fchollet — COMMENTED** [review]
Thank you for the PR! Looking great!

File: examples/nlp/summarization.py (hunk 2 above)
Comment (fchollet) [reply examples/nlp/summarization.py:330]: Did you figure this out?

Comment (Sreyan88) [reply examples/nlp/summarization.py:330]: Hi @fchollet ,

First of all, thank you so much for your detailed feedback and review on all my PRs. This looks a bit tricky as passing Rouge from KerasNLP metric to `compile()` would require my model inputs and targets to be strings which is not the case in my current implementation (my current implementation works on input ids). So if I want to use Rouge from KerasNLP metric, I have two options:

1) Simply replace [this line](https://github.com/keras-team/keras-io/pull/967/files#diff-2deb36c911c9b27529bce55cb6f0729bbb9d0f27a0ba0c0f137659538ae3fbe2R310-R312) with rouge from KerasNLP. This currently uses rouge from Hugging Face. This will keep the callback as it is.

2) Write a custom evaluation metric that decodes my model to strings similar and use KerasNLP Rouge in that metric, something similar to my WER implementation [here](https://github.com/keras-team/keras-io/pull/996/files#diff-e8a623de2815fff9118c26111ecf0fa511df13fb0eb8b19222f1172aa2aac28dR594-R603).

File: examples/nlp/summarization.py (hunk 1 above)
Comment (Sreyan88) [reply examples/nlp/summarization.py]: Hi @fchollet ,

I just realized that I might need to revert this change. My model isn't training well after this, and the loss doesn't decrease ( I just changed the optimizer, nothing else). I will revert this change.

**Comment by Sreyan88** [comment]
Hi @fchollet ,

Thank You for the comments. I have made some changes to the evaluation metric and optimizer, please let me know if this suits well to needs!

**Comment by fchollet** [comment]
I pushed some copyedits, please pull them first. You can now add the generated files. Sorry for the delay!

**Comment by Sreyan88** [comment]
Hi @fchollet ,

Done adding the files! Thank You again for the time!

**Review by fchollet — APPROVED** [review]
Thank you!
